includes/snmp.inc.php

```diff
@@ -130,7 +130,19 @@ function gen_snmpget_cmd($device, $oids, $options = null, $mib = null, $mibdir =
  */
 function gen_snmpwalk_cmd($device, $oids, $options = null, $mib = null, $mibdir = null)
 {
-    if ($device['snmpver'] == 'v1' || (isset($device['os']) && Config::getOsSetting($device['os'], 'snmp_bulk', true) == false)) {
+    // look for MIBs which need to use slow snmpwalk
+    $flwalk = false;
+    $cnfsnmp = Config::getOsSetting($device['os'], 'snmp');
+    if (isset($cnfsnmp['no_bulk'])) {
+        foreach ($cnfsnmp['no_bulk'] as $wkey => $mnames) {
+            if (Str::contains(strtoupper($mib), strtoupper($mnames))) {
+                $flwalk = true;
+                d_echo("\n" . 'SNMP: ' . $mib . ' forced to snmpwalk');
+            }
+        }
+    }
+
+    if ($device['snmpver'] == 'v1' || (isset($device['os']) && Config::getOsSetting($device['os'], 'snmp_bulk', true) == false) || $flwalk) {
```

Comment:
`|| $flwalk` seems to be different from the logic from NetSnmpQuery

Reply:
var names and SnmpQuery logic modified
hope now is OK